opm/io/eclipse/OutputStream.hpp

```diff
@@ -0,0 +1,175 @@
+
+        const ResultSet& resultSetDescriptor() const
+        {
+            return this->rset_;
+        }
+
+        bool formatted() const
+        {
+            return this->formatted_;
+        }
+
+        bool unified() const
+        {
+            return this->unified_;
+        }
+
+        /// Access writable output stream.
+        ///
+        /// Must not be called prior to \c prepareStep.
+        EclOutput& stream();
+
+    private:
+        ResultSet rset_;
+        bool      formatted_;
+        bool      unified_;
+
+        /// Restart output stream.
+        std::unique_ptr<EclOutput> stream_;
+
+        /// Open unified output file and place stream's output indicator
+        /// in appropriate location.
+        ///
+        /// Writes to \c stream_.
+        ///
+        /// \param[in] fname Filename of output stream.
+        ///
+        /// \param[in] seqnum Sequence number of new report.  One-based
+        ///    report step ID.
+        void openUnified(const std::string& fname,
+                         const int          seqnum);
+
+        /// Open new output stream.
+        ///
+        /// Handles the case of separate output files or unified output file
+        /// that does not already exist.  Writes to \c stream_.
+        ///
+        /// \param[in] fname Filename of new output stream.
+        void openNew(const std::string& fname);
+
+        /// Open existing output file and place stream's output indicator
+        /// in appropriate location.
+        ///
+        /// Writes to \c stream_.
+        ///
+        /// \param[in] fname Filename of output stream.
+        ///
+        /// \param[in] writePos Position at which to place stream's output
+        ///    indicator.  Use \code streampos{ streamoff{-1} } \endcode to
+        ///    place output indicator at end of file (i.e, simple append).
+        void openExisting(const std::string&   fname,
+                          const std::streampos writePos);
+    };
+
+    /// Derive filename corresponding to output stream of particular result
+    /// set, with user-specified file extension.
+    ///
+    /// Low-level string concatenation routine that does not know specific
+    /// relations between base names and file extensions.  Handles details
+    /// of base name ending in a period (full stop) or having a name that
+    /// might otherwise appear to contain a file extension (e.g., CASE.01).
+    ///
+    /// \param[in] rsetDescriptor Output directory and base name of result set.
+    ///
+    /// \param[in] ext Filename extension.
+    ///
+    /// \return outputDir/baseName.ext
+    std::string outputFileName(const ResultSet&   rsetDescriptor,
+                               const std::string& ext);
+
+    /// Derive filename of restart output stream.
+    ///
+    /// Knows the rules for formatted vs. unformatted, and separate vs.
+    /// unified output files.
+    ///
+    /// \param[in] rst Previously configured restart stream.
```

Comment:
Please cut down on these comments - it is just noise when my eyes are searching for the real content; i.e. the declarations.